src/west/manifest.py

```diff
@@ -248,6 +248,45 @@ def _load(self, data, sections):
         self._remotes_dict = remotes_dict
         self.projects = tuple(projects)
 
+    def dump(self, sections):
+        data = {}
+        if 'west' in sections:
+            w = {'url' : self.west_project.url,
+                 'revision' : self.west_project.revision}
+            data['west'] = w
+        if 'manifest' in sections:
+            m = {}
+            sm = self._data['manifest']
+            # Add the optional defaults section only if present in the
+            # original manifest. It is not overridable so just use the
+            # original data if required.
+            if 'defaults' in sm:
+                m['defaults'] = sm['defaults']
+
+            r = []
+            for remote in self.remotes:
+                r.append({'name' : remote.name, 'url' : remote.url})
+            m['remotes'] = r
+
+            p = []
+            for project in self.projects:
+                lp = {}
+                sp = [i for i in sm['projects'] if i['name'] == project.name][0]
+                lp['name'] = project.name
```

Comment:
Sorry, I'm confused about why all this is necessary. Can't we just properly format `project` as a dict and keep going?

Is the idea to try to "minify" the dumped representation?